apps/web/src/components/virtualized-tree/index.tsx

```diff
@@ -120,8 +136,7 @@ export function VirtualizedTree<T>(props: TreeViewProps<T>) {
           return;
         }
 
-        // TODO: double check
-        if (node.hasChildren) {
+        if (itemOptions?.expand) {
```

Comment:
What if the node doesn't have children?

Reply:
The `node.hasChildren` check to decide whether to expand the node is faulty. I think I added it way back but never came back to it again. 

`node.hasChildren` check assumes that the node should always be expanded if it has children. Whereas we only want to expand when a subnotebook is created against the node. `refreshItem` doesn't have enough information/context to check for this case, so I'm passing it in as a param.

Reply:
Yes, but we should still check it in addition to `itemOptions.expand`.

Reply:
Done, I have to move the check after we have fetched the latest children of the node otherwise `node.hasChildren` doesn't have updated information

Reply:
> I have to move the check after we have fetched the latest children

But that way the notebook won't expand because `expandedIds` would be updated after we have fetched the children.

Reply:
It does expand because we update `expandedIds` before it is passed in `setExpandedIds` as updated state

Reply:
Yes but `expandedIds` is also used in `fetchChildren` for prefetching a node's children if its expanded.